packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -242,35 +242,43 @@ export class AssetCode extends Code {
   }
 
   public bind(scope: cdk.Construct): CodeConfig {
-    // If the same AssetCode is used multiple times, retain only the first instantiation.
-    if (!this.asset) {
-      this.asset = new s3_assets.Asset(scope, 'Code', {
+    const stackPath = cdk.Stack.of(scope).node.path; // use path instead of name to account for Stage.
```

Comment:
Is this good enough? Wouldn't the same stack path in multiple `Apps` still give the same error?

Reply:
Aha well. This change DOES address the issue it's purporting to address (use same code in multiple stacks inside the same app), but it does NOT address the error you mentioned, and would still fail if you reused a single `Code` object among multiple unit tests.

```
Cannot reference across apps. Consuming and producing stacks must be defined within the same CDK app.
```